Realm/RLMClassInfo.mm

```diff
@@ -78,6 +78,11 @@
     return *m_linkTargets[propertyIndex];
 }
 
+RLMClassInfo &RLMClassInfo::linkTargetType(realm::Property const& property) {
+    REALM_ASSERT(property.type == PropertyType::Object || property.type == PropertyType::Array);
```

Comment:
None of the places where `linkTargetType()` is used are operations applicable to backlinks.